edxval/tests/test_api.py

```diff
@@ -1333,37 +1380,158 @@ def test_invalid_course_id(self):
+            self.file_system,
+            constants.EXPORT_IMPORT_STATIC_DIR
+        )
         self.assertEqual(
             VideoTranscript.objects.count(),
             0
         )
 
+    def test_import_transcript_attached_existing_video(self):
+        """
+        Verify that transcript import for existing video with transcript attached is working as expected.
+        """
+        expected_video_transcripts = [self.transcript_data3]
+
+        import_xml = self.make_import_xml(
+            video_dict=constants.VIDEO_DICT_FISH,
+            video_transcripts=expected_video_transcripts
+        )
+
+        # Verify video is present before.
+        video = Video.objects.get(edx_video_id=constants.VIDEO_DICT_FISH['edx_video_id'])
+        self.assertIsNotNone(video)
+
+        # Create internal video transcripts
+        transcript_data = dict(constants.VIDEO_TRANSCRIPT_3PLAY, video=video)
+        transcript_data.pop('video_id')
+        transcript_data.pop('file_data')
+        transcript = VideoTranscript.objects.create(**transcript_data)
+
+        # Verify that video has expected transcripts before import.
+        self.assert_transcripts(
+            constants.VIDEO_DICT_FISH['edx_video_id'],
+            expected_video_transcripts
+        )
+
+        api.import_from_xml(
+            import_xml,
+            constants.VIDEO_DICT_FISH['edx_video_id'],
+            self.file_system,
+            constants.EXPORT_IMPORT_STATIC_DIR,
+            'test_course_id'
+        )
+
+        # Verify that video has expected transcripts after import.
+        self.assert_transcripts(
+            constants.VIDEO_DICT_FISH['edx_video_id'],
+            expected_video_transcripts
+        )
+
+    def test_import_no_transcript_attached_existing_video(self):
+        """
+        Verify that transcript import for existing video with no transcript attached is working as expected.
+        """
+        exported_video_transcripts = [self.transcript_data1, self.transcript_data2]
+
+        # Verify video is present before.
+        video = Video.objects.get(edx_video_id=constants.VIDEO_DICT_FISH['edx_video_id'])
+        self.assertIsNotNone(video)
+
+        import_xml = self.make_import_xml(
+            video_dict=constants.VIDEO_DICT_FISH,
+            video_transcripts=exported_video_transcripts
+        )
+
+        # There must not be any transcript before import.
+        with self.assertRaises(VideoTranscript.DoesNotExist):
+            VideoTranscript.objects.get(video__edx_video_id=constants.VIDEO_DICT_FISH['edx_video_id'])
+
+        api.import_from_xml(
+            import_xml,
+            constants.VIDEO_DICT_FISH['edx_video_id'],
+            self.file_system,
+            constants.EXPORT_IMPORT_STATIC_DIR,
+            'test_course_id'
+        )
+
+        # Verify that no transcript record is created.
+        self.assert_transcripts(
+            constants.VIDEO_DICT_FISH['edx_video_id'],
+            []
+        )
```

Comment:
why about using the below instead of `assert_transcripts`?
```python
with self.assertRaises(VideoTranscript.DoesNotExist):
    VideoTranscript.objects.get(video__edx_video_id=constants.VIDEO_DICT_FISH['edx_video_id'])
```